### DISCUSSION TITLE
image->save default permissions

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [phalcon/cphalcon#16620] [General]
this script correctly writes a file with the OS inherited permissions.

```
<?php

$filename = 'test-file.txt';

$folder = getcwd();
$perms = substr(sprintf('%o', fileperms($folder)), -4);
echo "$perms";

touch($filename);
```

However my phalcon is writing with different mode
`$image->save($path, $quality);`

Is this something default in phalcon? (php mkdir also has something weird for default)

**Comment by f1-outsourcing** [comment]
I have issues getting the parent dir permissions, remove execute rights and use that mode for files. Don't really get what fileperms is outputting.

```
            $perms = 0770;
            $perms = $perms & 0777;  // remove the leading crap
            $perms = $perms & ~0111; // remove execute
            chmod($path,$perms);
```

**Comment by niden** [comment]
What do you use as a class to call `save()`. I see your variable is `$image`. Is that a `Phalcon\Image\*` adapter?

**Reply by f1-outsourcing** [reply]
$image = new \Phalcon\Image\Adapter\GD("{$filePath}/{$fileName}");

I can't reproduce this any more. I think it was related to running php-fpm without the proper group (instead of user apache)  (was having a dir g+s)
